src/securemul.rs

```diff
@@ -0,0 +1,430 @@
+use crate::error::BoxError;
+use crate::field::Field;
+use crate::helpers::ring::{HelperAddr, Ring};
+use crate::prss::Participant;
+use crate::replicated_secret_sharing::ReplicatedSecretSharing;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use thiserror::Error;
+
+/// Secure multiplication protocol using replicated secret sharing over some field `F`.
+/// * [paper](https://eprint.iacr.org/2018/387.pdf)
```

Comment:
/// IKHC multiplication protocol
/// for use with replicated secret sharing over some field `F`.
/// K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, and B. Pinkas. High-throughput secure AES computation. In WAHC@CCS 2018, pp. 13–24, 2018